62.CAD/common.hlsl

```diff
@@ -28,27 +28,24 @@ struct QuadraticBezierInfo
     // TODO[Przemek]: Any Data related to precomputing things for beziers will be here
 };
 
-// TODO[Lucas]:
-/*
-You need a struct here that represents a curve which is referenced by a "CurveBox"
-Which is basically the same as QuadraticBezierInfo, if the middle point is "nan" it means it's a line connected by p0 and p2
-*/
-
-// TODO[Lucas]:
-/*
-You need another struct here that represents a "CurveBox" which
-0. have a aabb (double2 min, max) which will get transformed in the vertex shader, and will be calculated on the cpu when generating these boxes
-1. references two Curves by `uint64_t address` into the geometry buffer
-2. It will also contain tmin,tmax for both curves (becuase we subdivide curves into smaller monotonic parts we need this info to help us discard invalid solutions)
-*/
+struct CurveBox 
+{
+    // will get transformed in the vertex shader, and will be calculated on the cpu when generating these boxes
+    double2 aabbMin;
+    double2 aabbMax; // 32
+    double2 curveMin[3]; // 80
+    double2 curveMax[3]; // 128
+};
 
 struct Globals
 {
     double4x4 viewProjection; // 128 
     double screenToWorldRatio; // 136 - TODO: make a float, no point making it a double
     uint2 resolution; // 144
     float antiAliasingFactor; // 148
-    float _pad; // 152
+    int clipEnabled; // 152
+    int2 _pad; // 164
+    double4 clip; // 192
```

Comment:
remove all the clipping stuff. you did for n4ce and brought here, it was wrong and will be changed